OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/usb/CcidTransceiver.java

```diff
@@ -0,0 +1,169 @@
+                0x00,
+                mCounter++,
+                0x00,
+                0x00, 0x00
+        };
+
+        sendRaw(iccPowerCommand);
+
+        long startTime = System.currentTimeMillis();
+        byte[] atr = null;
+        while (true) {
+            try {
+                atr = receiveRaw();
+                break;
+            } catch (Exception e) {
+                // Try more startTime
+                if (System.currentTimeMillis() - startTime > TIMEOUT) {
+                    break;
+                }
+            }
+            SystemClock.sleep(100);
+        }
+
+        if (atr == null) {
+            throw new UsbTransportException("Couldn't power up Security Token");
+        }
+
+        return atr;
+    }
+
+    /**
+     * Transmits XfrBlock
+     * 6.1.4 PC_to_RDR_XfrBlock
+     * @param payload payload to transmit
+     * @throws UsbTransportException
+     */
+    public void sendXfrBlock(byte[] payload) throws UsbTransportException {
+        int l = payload.length;
+        byte[] data = Arrays.concatenate(new byte[]{
+                        0x6f,
+                        (byte) l, (byte) (l >> 8), (byte) (l >> 16), (byte) (l >> 24),
+                        0x00,
+                        mCounter++,
+                        0x00,
+                        0x00, 0x00},
+                payload);
+
+        int send = 0;
+        while (send < data.length) {
+            final int len = Math.min(mBulkIn.getMaxPacketSize(), data.length - send);
+            sendRaw(Arrays.copyOfRange(data, send, send + len));
+            send += len;
+        }
+    }
+
+    public byte[] receive() throws UsbTransportException {
+        byte[] buffer = new byte[mBulkIn.getMaxPacketSize()];
+        byte[] result = null;
+        int readBytes = 0, totalBytes = 0;
+
+        do {
+            int res = mConnection.bulkTransfer(mBulkIn, buffer, buffer.length, TIMEOUT);
+            if (res < 0) {
+                throw new UsbTransportException("USB error - failed to receive response " + res);
+            }
+            if (result == null) {
+                if (res < 10) {
+                    throw new UsbTransportException("USB-CCID error - failed to receive CCID header");
+                }
+                totalBytes = ByteBuffer.wrap(buffer, 1, 4).order(ByteOrder.LITTLE_ENDIAN).asIntBuffer().get() + 10;
+                result = new byte[totalBytes];
+            }
+            System.arraycopy(buffer, 0, result, readBytes, res);
+            readBytes += res;
+        } while (readBytes < totalBytes);
+
+        return result;
+    }
+
+    private void sendRaw(final byte[] data) throws UsbTransportException {
+        final int tr1 = mConnection.bulkTransfer(mBulkOut, data, data.length, TIMEOUT);
+        if (tr1 != data.length) {
+            throw new UsbTransportException("USB error - failed to transmit data " + tr1);
+        }
+    }
+
+    private byte getStatus(byte[] bytes) {
```

Comment:
static, and could also use a short comment on what's going on here (like the other methods refer to relevant specs)